lib/broadway/producer.ex

```diff
@@ -47,11 +47,20 @@ defmodule Broadway.Producer do
   def init({args, index}) do
     {module, arg} = args[:module]
     transformer = args[:transformer]
+    n_processors = args[:n_processors]
+    processors_partition_by = args[:processors_partition_by]
+    processors_name = args[:processors_name]
+    batchers_names = args[:batchers_names]
+    broadway_name = args[:broadway_name]
 
     # Inject the topology index only if the args are a keyword list.
     arg =
       if Keyword.keyword?(arg) do
-        Keyword.put(arg, :broadway_index, index - 1)
+        arg
+        |> Keyword.put(:broadway_index, index - 1)
+        |> Keyword.put(:processors_name, processors_name)
+        |> Keyword.put(:batchers_names, batchers_names)
+        |> Keyword.put(:broadway_name, broadway_name)
```

Comment:
What should we do when arg is not a keyword list? Maybe changing the API and pass another argument?

Reply:
What if we put all of this information inside a single `:broadway` key?